### DISCUSSION TITLE
Amazon music

### DISCUSSION THREAD

**Discussion by properlypurple** [music-assistant/support#626] [Feature requests and ideas]
### Current Status
As at August 2025 discussions were had with Amazon but a technical roadblock was encountered due to their DRM requirements. For this reason no work is currently underway for this provider.

### Original Post 
It would be great to have Amazon prime music as a provider. I found a python library that can help as an example
https://github.com/Jaffa/amazon-music

**Comment by tombloemker** [comment]
Doesn't everyone have Prime?

**Comment by Bettiol** [comment]
Any news on the Amazon Music integration?Thanks

**Reply by OzGav** [reply]
No news sorry. We would need a new dev to pick this up.

**Comment by HiChenJo** [comment]
Amazon is working on a new API. 
https://developer.amazon.com/de/docs/music/API_web_overview.html

**Reply by mterry63** [reply]
Looks like the API is in a closed beta now https://developer.amazon.com/docs/music/API_web_overview.html

**Reply by TicoM1** [reply]
Their API 1.0 website still says it's in closed beta. Has anyone heard news if they ever plan to release it?

**Comment by nocturnalarchives** [comment]
Any chance this might happen soon?

**Reply by OzGav** [reply]
This is not currently being worked on, Needs a developer to pick it.

**Comment by Alulu2401** [comment]
hi all  :) 
small up on this feature. It would be great to have this one :D 
Anyway thanks for this hard work !

**Comment by mac-city** [comment]
I would love to see amazon music and audible!

**Comment by tonyspence** [comment]
Unfortunately without Amazon Prime Music this product is of little or no use to me. Will keep monitoring in the hope of something materialising during what remains of my lifetime. I will just have to continue using the execrable Sonos software

**Reply by OzGav** [reply]
You can submit a PR adding this at anytime.

**Reply by trainman** [reply]
Are there any docs that'll help someone who hasn't contributed to Music Assistant understand the interface they are building for and how to test their code from Music Assistant? I read through the code in the Deezer and SoundCloud repos, and it looks easy enough. I can't find any documentation for getting started as a contributor. Yes, I tried Google. I don't do discord or need another chat app. I'm already in too many Slack groups. 
https://www.music-assistant.io/help/

**Reply by OzGav** [reply]
It is preferable that you join Discord to discuss the development and then to provide assistance to users once your provider is merged. Here are some links for you to get started 

https://github.com/music-assistant/server/blob/dev/DEVELOPMENT.md

https://github.com/music-assistant/server/tree/dev/music_assistant/providers/_template_music_provider

**Reply by trainman** [reply]
Perfect, thank you!

**Comment by tombloemker** [comment]
I requested Amazon music support over a year ago, I was under the
impression their non-standard stream made in nearly impossible to support?

**Reply by OzGav** [reply]
It’s irrelevant how long ago you made a request. Someone needs to do the work. You need to make less sarcastic comments when no one does.

**Reply by StefanX13** [reply]
Don't know how tombloekers question is more sarcastic than your answer to  tonyspences question.

**Comment by DamageCase13** [comment]
i would seriously love for amazon to be added!

**Comment by robertkearns** [comment]
I’d be interested in taking this on. Has anyone done any work on this yet, or have an understanding of the Amazon specific challenges?

**Reply by tombloemker** [reply]
I think there is some info on the first comment in this thread. Also someone stated Amazon is currently changing the music API.  It would be good to clarify the objective. If you are wanting access to Amazon purchased digital media or to prime music subscription? Does Amazon differentiate between these? Would all content be in one repository? Are the formats different? 
(All comments are meant constructively and in no way meant sarcastically.)  😉

**Comment by mr-ransel** [comment]
I was just looking into this, and I suspect that based on Amazon's requirements for integration, this is going to be a non-starter, and they won't be issuing any beta auth tokens, and even if you extract one from an approved player, I'm betting they'd find a way to revoke it:
Requirement 2.b. from their list of [requirements](https://developer.amazon.com/docs/music/requ_AM-Program-Requirements.html) is
> Your Music Product must not [...] integrate the Amazon Music Service or any Amazon Music Service content with a third-party music service or any third-party music service content;

Isn't this the purpose of Music Assistant? To tie together libraries with other services and collate their metadata together?

And even if we did meet their approval, I don't think they meet ours. Requirement 3.a and 3.e:
> In connection with including access to the Amazon Music Service in Your Music Product and allowing end users to access the Amazon Music Service through Your Music Product [...] you will implement in Your Music Product the collection of any usage metrics or related data that we may require and will report such metrics or data to us;

Well, I really don't think that's something we _want_ to integrate into Music Assistant, is it?

> you will not facilitate or provide access to the Amazon Music Service through any means other than through the authentication methods we specify, and you will not disable, circumvent or avoid any security device, mechanism or protocol of the Amazon Music Service;

Well, again, that's kind of the point of Music Assistant. To connect to the music library source, and provide access to it to other devices. This is arguable, since nothing is being circumvented, and could be argued that it's just streaming the music to speakers, like any other bluetooth speaker.

Most of the above is apparent in the way their API is structured; in several places it's designed to control what buttons can be pressed when, and making sure all button press information makes it's way back to Amazon so I don't think there is a good path towards implementation for this in an open source product like Music Assistant, unfortunately.

**Reply by tombloemker** [reply]
This is a good reason to discontinue any relationship with Amazon music. Maybe this explanation should be posted with a recommended merchant list that has friendly APIs.

**Reply by disconn3ct** [reply]
> recommended merchant list

A list of vendors that are not supported in the slightest even a little, but are "nice"? That seems off-topic.

**Reply by joker11940** [reply]
> > recommended merchant list
> 
> A list of vendors that are not supported in the slightest even a little, but are "nice"? That seems off-topic. 

Of topic, yes, but honestly, it would have saved me the 30 minutes of digging to get here. Jut have a a not-supported list below the normal list and a reason for why.

I agree with all the points here but it does not change my current situation, I may look to move away from Amazon in the future but there is no incentive for me to move to Music Assistant in that case (I get that's not their problem 😅).

If I move to something like Spotify I believe Home Assistant supports that natively.. (I think 😅)

IDK, I guess it's just a sucky situation, right? Come on Amazon play nice, share your toys, keep people like me as a customer.

**Reply by trainman** [reply]
When I contacted the Amazon Dev Relations team, they liked the idea of someone integrating Amazon Music into Music Assistant. Unfortunately, I asked late, and the Beta is currently full. They said they should have more slots available in mid-2025.

Re: the points above.

2.b is now 2.e: It's clear that they don't want their content re-streamed via another service. They are not against integration into music assistant. If 2.e had this meaning, they wouldn't be OK with the Sonos integration. Amazon Music is one of 2 providers with the highest supported fidelity HD music on Sonos.

3.a and 3.e: There are no methods w/in the API to support either of these. It appears "a" is a legal catch-all to cover them capturing usage metrics. To me, "e" looks like something to say consumers can't skip ad content if we push it.

I don't understand why the following is highlighted. It's common sense that someone doing the following isn't following the rules.
> you will not facilitate or provide access to the Amazon Music Service through any means other than through the authentication methods we specify, and you will not disable, circumvent or avoid any security device, mechanism or protocol of the Amazon Music Service;

I see nothing in the October 29, 2024, program requirements that seem out of place or evil. I've purchased millions in annualized commercial software contracts and sold millions in prior roles. As a tech executive and developer, I find legal terms sound weird. After reading too many documents and working with too many lawyers over the years, none of this is language I am surprised by or cause for concern. (again, I'm not an attorney, and this isn't legal advice)

**Comment by HumanBot000** [comment]
Has anyone started working on this?

**Reply by OzGav** [reply]
We have had discussions with Amazon but they encrypt their streams and we couldn’t get agreement as to a solution that would work for both of us. At some point we will revisit this.

**Comment by joker11940** [comment]
So we now have Amazon Audible but not Amazon Music. Did anyone revisit this? I know Amazon isn't everyone's favourite company but they are one of the most popular providers. Getting this even with limited functionality would allow a lot of people to kickstart it into Music Assistant.

**Reply by OzGav** [reply]
Did you read the first post?
